src/TensorOneD.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <math.h>
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+TensorOneD::TensorOneD()
+{
+}
+
+TensorOneD::TensorOneD(int dim)
+{
+    set_dim(dim);
+}
+
+TensorOneD::TensorOneD(const TensorOneD &other)
+{
+    set_dim(other.get_dim());
+    for (int idx = 0; idx < (int)m_vec.size(); idx++) {
+        m_vec[idx] = other[idx];
+    }
```

Comment:
I believe you can replace this with: ``m_vec = other.m_vec`` (considering ``std::vector`` implements ``operator=()`` and considering the vector is of a primitive type).

However, also considering the default implementations of the copy constructor and assignment operator, I believe you should be able to get away with not defining them all together.